insights/client/support.py

```diff
@@ -37,12 +36,8 @@ def _legacy_registration_check(pconn):
 
     api_reg_status = pconn.api_registration_check()
     logger.debug('Registration status: %s', api_reg_status)
-    if type(api_reg_status) is bool:
-        if api_reg_status:
-            api_record = 'Insights API confirms registration.'
-        else:
-            api_record = 'Insights API could not be reached to confirm registration status.'
-            unreachable = True
+    if api_reg_status is True:
+        api_record = 'Insights API confirms registration.'
     elif api_reg_status is None:
         api_record = 'Insights API says this machine is NOT registered.'
         api_reg_status = False
```

Comment:
This overwrite of the value retrieved from _api_registration_check_ is pretty ugly, or at least non-semantic.